### ISSUE TITLE
Crash on enable ignoring spaces in diff

### ISSUE THREAD

**Issue by MatkovIvan** [ForkIssues/Tracker#1055]
```
Process:               Fork [92144]
Path:                  /Applications/Fork.app/Contents/MacOS/Fork
Identifier:            Fork
Version:               1.0.93.6 (1.0.93.6)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Fork [92144]
User ID:               501

Date/Time:             2020-06-16 11:34:18.517 +0300
OS Version:            Mac OS X 10.15.5 (19F101)
Report Version:        12
Bridge OS Version:     3.0 (14Y908)
Anonymous UUID:        8D68EC05-BB21-2E6F-6E08-4D19F8D609E9

Sleep/Wake UUID:       531955AD-EA05-437C-9240-EFF7AD3AAEE0

Time Awake Since Boot: 200000 seconds
Time Since Wake:       3100 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes:       0x0000000000000001, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Illegal instruction: 4
Termination Reason:    Namespace SIGNAL, Code 0x4
Terminating Process:   exc handler [92144]

Application Specific Information:
Performing @selector(ignoreWhitespacesToggleButtonClicked:) from sender NSButton 0x7f957ee6e680

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   com.DanPristupov.Fork         	0x00000001045c1605 0x1042d0000 + 3085829
1   com.DanPristupov.Fork         	0x00000001045c154b 0x1042d0000 + 3085643
2   com.apple.CoreFoundation      	0x00007fff34fb589f __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
3   com.apple.CoreFoundation      	0x00007fff34fb5833 ___CFXRegistrationPost1_block_invoke + 63
4   com.apple.CoreFoundation      	0x00007fff34fb57a8 _CFXRegistrationPost1 + 372
5   com.apple.CoreFoundation      	0x00007fff34fb5414 ___CFXNotificationPost_block_invoke + 80
6   com.apple.CoreFoundation      	0x00007fff34f8558d -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1554
7   com.apple.CoreFoundation      	0x00007fff34f84a39 _CFXNotificationPost + 1351
8   com.apple.Foundation          	0x00007fff375ff786 -[NSNotificationCenter postNotificationName:object:userInfo:] + 59
9   com.DanPristupov.Fork         	0x00000001043fef54 0x1042d0000 + 1240916
10  com.DanPristupov.Fork         	0x00000001043fe368 0x1042d0000 + 1237864
11  com.apple.AppKit              	0x00007fff32476fc7 -[NSApplication(NSResponder) sendAction:to:from:] + 299
12  com.apple.AppKit              	0x00007fff32476e62 -[NSControl sendAction:to:] + 86
13  com.apple.AppKit              	0x00007fff32476d94 __26-[NSCell _sendActionFrom:]_block_invoke + 136
14  com.apple.AppKit              	0x00007fff32476c96 -[NSCell _sendActionFrom:] + 171
15  com.apple.AppKit              	0x00007fff32476bdd -[NSButtonCell _sendActionFrom:] + 96
16  com.apple.AppKit              	0x00007fff32472ebb NSControlTrackMouse + 1745
17  com.apple.AppKit              	0x00007fff324727c2 -[NSCell trackMouse:inRect:ofView:untilMouseUp:] + 130
18  com.apple.AppKit              	0x00007fff32472681 -[NSButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 691
19  com.apple.AppKit              	0x00007fff324719fd -[NSControl mouseDown:] + 748
20  com.apple.AppKit              	0x00007fff3246fe10 -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 4914
21  com.apple.AppKit              	0x00007fff323da611 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 2612
22  com.apple.AppKit              	0x00007fff323d99b9 -[NSWindow(NSEventRouting) sendEvent:] + 349
23  com.apple.AppKit              	0x00007fff323d7d44 -[NSApplication(NSEvent) sendEvent:] + 352
24  com.apple.AppKit              	0x00007fff32224daf -[NSApplication run] + 707
25  com.apple.AppKit              	0x00007fff321f6b86 NSApplicationMain + 777
26  com.DanPristupov.Fork         	0x00000001042d1ff9 0x1042d0000 + 8185
27  libdyld.dylib                 	0x00007fff6efbacc9 start + 1

Thread 1:
0   libsystem_kernel.dylib        	0x00007fff6f0fbdfa mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6f0fc170 mach_msg + 60
2   com.DanPristupov.Fork         	0x000000010476b08a 0x1042d0000 + 4829322
3   libsystem_pthread.dylib       	0x00007fff6f1bf109 _pthread_start + 148
4   libsystem_pthread.dylib       	0x00007fff6f1bab8b thread_start + 15

Thread 2:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	0x00007fff6f0fbdfa mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6f0fc170 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff34fbff85 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff34fbea52 __CFRunLoopRun + 1319
4   com.apple.CoreFoundation      	0x00007fff34fbdece CFRunLoopRunSpecific + 462
5   com.apple.AppKit              	0x00007fff323d6144 _NSEventThread + 132
6   libsystem_pthread.dylib       	0x00007fff6f1bf109 _pthread_start + 148
7   libsystem_pthread.dylib       	0x00007fff6f1bab8b thread_start + 15

Thread 3:: com.apple.NSURLConnectionLoader
0   libsystem_kernel.dylib        	0x00007fff6f0fbdfa mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6f0fc170 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff34fbff85 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff34fbea52 __CFRunLoopRun + 1319
4   com.apple.CoreFoundation      	0x00007fff34fbdece CFRunLoopRunSpecific + 462
5   com.apple.CFNetwork           	0x00007fff337f802a 0x7fff337ae000 + 303146
6   com.apple.Foundation          	0x00007fff3764e7a2 __NSThread__start__ + 1064
7   libsystem_pthread.dylib       	0x00007fff6f1bf109 _pthread_start + 148
8   libsystem_pthread.dylib       	0x00007fff6f1bab8b thread_start + 15

Thread 4:: JavaScriptCore bmalloc scavenger
0   libsystem_kernel.dylib        	0x00007fff6f0fe882 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff6f1bf425 _pthread_cond_wait + 698
2   libc++.1.dylib                	0x00007fff6c28e592 std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&) + 18
3   com.apple.JavaScriptCore      	0x00007fff397ba224 void std::__1::condition_variable_any::wait<std::__1::unique_lock<bmalloc::Mutex> >(std::__1::unique_lock<bmalloc::Mutex>&) + 84
4   com.apple.JavaScriptCore      	0x00007fff397bea2b bmalloc::Scavenger::threadRunLoop() + 299
5   com.apple.JavaScriptCore      	0x00007fff397be5f9 bmalloc::Scavenger::threadEntryPoint(bmalloc::Scavenger*) + 9
6   com.apple.JavaScriptCore      	0x00007fff397c0cd7 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(bmalloc::Scavenger*), bmalloc::Scavenger*> >(void*) + 39
7   libsystem_pthread.dylib       	0x00007fff6f1bf109 _pthread_start + 148
8   libsystem_pthread.dylib       	0x00007fff6f1bab8b thread_start + 15

Thread 5:: com.apple.coreanimation.render-server
0   libsystem_kernel.dylib        	0x00007fff6f0fbdfa mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6f0fc170 mach_msg + 60
2   com.apple.QuartzCore          	0x00007fff40acbf7e CA::Render::Server::server_thread(void*) + 496
3   com.apple.QuartzCore          	0x00007fff40acbd87 thread_fun(void*) + 25
4   libsystem_pthread.dylib       	0x00007fff6f1bf109 _pthread_start + 148
5   libsystem_pthread.dylib       	0x00007fff6f1bab8b thread_start + 15

Thread 6:
0   libsystem_pthread.dylib       	0x00007fff6f1bab68 start_wqthread + 0

Thread 7:
0   libsystem_pthread.dylib       	0x00007fff6f1bab68 start_wqthread + 0

Thread 8:
0   libsystem_pthread.dylib       	0x00007fff6f1bab68 start_wqthread + 0

Thread 9:
0   libsystem_pthread.dylib       	0x00007fff6f1bab68 start_wqthread + 0

Thread 10:
0   libsystem_pthread.dylib       	0x00007fff6f1bab68 start_wqthread + 0

Thread 11:
0   libsystem_pthread.dylib       	0x00007fff6f1bab68 start_wqthread + 0

Thread 12:
0   libsystem_pthread.dylib       	0x00007fff6f1bab68 start_wqthread + 0

Thread 13:
0   libsystem_pthread.dylib       	0x00007fff6f1bab68 start_wqthread + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x00007ffeeb92e250  rcx: 0x0000000100002f80  rdx: 0x00007fff6de2a448
  rdi: 0x00007f957d999750  rsi: 0x00000000000120a8  rbp: 0x00007ffeeb92e240  rsp: 0x00007ffeeb92e230
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000600002cc3338  r11: 0x0000600002cc3330
  r12: 0x00007f957d9996d0  r13: 0x00007f957d999750  r14: 0x00007fff881e4b60  r15: 0x0000600001714a80
  rip: 0x00000001045c1605  rfl: 0x0000000000010246  cr2: 0x0000600002cc3338
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     6
```

**Comment by DanPristupov** [comment]
I still don't know a way to reproduce this. The crash caused by a tricky memory leak which keeps the tab content alive after tab is closed.

**Comment by MatkovIvan** [comment]
I cannot reproduce it, it happened only once. So, I don't have reproduction steps too

btw thanks for quick reply

**Comment by DanPristupov** [comment]
I could not catch this crash for almost 2 (!) years and surprisingly just after you created this issue today I absolutely occasionally found what causes it 😀.

The steps to reproduce are like following:

- Have two open tabs
- Open the local changes view in one of them and stage a file using context menu
- Close the active tab
- Click on any icon on above the diff view in the remaining tab (e.g. 'ignore whitespaces').

The crash will be fixed in the next update (1.0.95).
